test/RNGCoordinator.test.js

```diff
@@ -0,0 +1,149 @@
+const {
+  buidler,
+  ethers,
+  expect,
+  getTestUsers,
+  callMultiReturnTx,
+  deployMockContract,
+  TEST_TOKEN,
+} = require('../js-utils/testEnv')
+
+const {
+  VRF,
+  VDF,
+  txOverrides,
+  contractManager,
+  toWei,
+  toBytes32,
+} = require('../js-utils/deployHelpers')
+
+const RNGInterface = require('../build/RNGInterface.json')
+const _getContract = contractManager(buidler)
+
+const debug = require('debug')('ptv3:RNGCoordinator.test')
+
+
+describe('RNGCoordinator contract', function() {
+  let users, rngCoordinator, rngService, rngService2
+
+  beforeEach(async () => {
+    users = await getTestUsers()
+
+    debug('mocking RNGInterface...')
+    rngService = await deployMockContract(users.deployer, RNGInterface.abi, txOverrides())
+    rngService2 = await deployMockContract(users.deployer, RNGInterface.abi, txOverrides())
+
+    debug('deploying RNGCoordinator...')
+    rngCoordinator = await _getContract('RNGCoordinatorHarness', [])
+    await rngService.mock.getLastRequestId.returns(1)
+  })
+
+  describe('randomNumber()', () => {
+    it('should throw error when no RNG service attached', async () => {
+      await expect(rngCoordinator.randomNumber(1))
+        .to.be.revertedWith('RNGCoordinator/no-rng-service')
+    })
+  })
+
+  describe('addRngService()', () => {
+    it('should prevent adding invalid RNG Services', async () => {
+      await expect(rngCoordinator.addRngService(ethers.constants.AddressZero))
+        .to.be.revertedWith('RNGCoordinator/invalid-rng-service')
+    })
+
+    it('should only allow the Owner/Governor to add RNG Services', async () => {
+      // Non-Owner
+      await expect(rngCoordinator.connect(users.stranger).addRngService(rngService.address))
+        .to.be.revertedWith('Ownable: caller is not the owner')
+
+      // Owner
+      await expect(rngCoordinator.addRngService(rngService.address))
```

Comment:
Can we separate this into it's own test case?  I.e. "it should revert when a non-owner attempts to add"